src/core/ui.js

```diff
@@ -130,6 +134,30 @@ function createWarnButton(butName, arr, title) {
   ariaDecorate(button, ariaMap);
   return button;
 }
+window.addEventListener("load", () => {
+  ui.addCommand(`Close`, ui.closeModal, "Esc", "❌");
+  menuLength = menu.children.length;
+});
+
+menu.addEventListener("keydown", e => {
```

Comment:
This is close, but not quite right. It should be possible to trap the tabbing just based the fact that the menu is "open".... i.e., it tabbing tries to leave the menu with a shift, it focuses the last, otherwise, if it tries to leave via the last, it goes to first.

Reply:
Unable to get you @marcoscaceres .
 Thats what I have done. When it tries to leave the last via Tab, it reaches the first.
When It tries to leaves the first via Shift+Tab, it reaches the last

Reply:
Understood, the logic is right, but I think the code could be simplified. Maybe worth adapting:
https://hiddedevries.nl/en/blog/2017-01-29-using-javascript-to-trap-focus-in-an-element

Reply:
Again, this remains way overly specific and it interferes with the eventing model... here a generalized alternative I quickly slapped together (it's incomplete, but should give you an idea):

https://jsfiddle.net/j7205dsf/11/

What we are trying to get to is something that should work with any tree structure, menu or anything else, as it only cares if the focus has tried to break out of the container. It if it breaks out, the focus is returned to the first focusable element.

Reply:
(Please don't insert tabindex=0 into `li` elements, which will cause focusing a single item twice)

Reply:
Yes, what @saschanaz said. The code above is just to illustrate a somewhat generalized way of solving this. 

Again, the problem to solve is:

  * given _any_ element (menu, list, Iframe, whatever), how can I trap the focus inside it allowing people to tab around to any focusable element.
 
The same solution must be usable in both the ReSpec menu and any of the popups dialogs.

Reply:
Thanks for your help @marcoscaceres . I am working on a genric solution to this and come out with it soon!!

Reply:
Appreciate your efforts. This is a really hard problem and definitely not "good first issue" territory. I think we are getting closer to a solution however. It might be worth Googling a bit more, however as this has to be solved problem. We just need to adapt someone else's solution.

Reply:
> Again, the problem to solve is:
given any element (menu, list, Iframe, whatever), how can I trap the focus inside it allowing people to tab around to any focusable element.

I think that https://hiddedevries.nl/en/blog/2017-01-29-using-javascript-to-trap-focus-in-an-element is an excellent solution and I have implemented on it. Sending the first and last focussable elements are an excellent way offcourse.
Does this seem right to you